### PR TITLE
Add an impl of PhfHash for UniCase

### LINKED ISSUE #21: Support ASCII-case-insensitive keys
@SimonSapin wanted this for CSS parsing. It could also be useful for HTML parsing.  It's pervasive in the Web platform and has a much simpler definition than Unicode case folding.

### PR DESCRIPTION
**Pull request by Ryman**
Fixes #21

### CODE REVIEW

**Comment by Ryman** [comment]
:zap: friendly ping :zap:  @sfackler

**Comment by sfackler** [comment]
Oops, looks good to me.

File: phf_shared/src/lib.rs [hunk 1]
```diff
@@ -103,6 +106,14 @@ impl PhfHash for [u8] {
     }
 }
 
+#[cfg(feature = "unicase")]
+impl<S> PhfHash for unicase::UniCase<S>
+where unicase::UniCase<S>: Hash {
+    #[inline]
+    fn phf_hash<H: Hasher>(&self, state: &mut H) {
+        self.hash(state)
```
Comment (sfackler) [inline phf_shared/src/lib.rs]: Does UniCase's Hash impl depend on the size of `usize`? I seem to remember that `str`'s implementation writes its length first, for example. This'll break things when cross compiling.

Comment (Ryman) [inline phf_shared/src/lib.rs]: I don't think so, it [hashes a char at a time](https://github.com/seanmonstar/unicase/blob/c09ee88f30ba6cf25d6ed23de6929987e9f245a1/src/lib.rs#L103-L105) (so no allocation required) and doesn't include the length. Unicase doesn't have any guarantee to keep this implementation, but I think it'd be breaking for them to change it in future so would be a version bump for them? cc @seanmonstar

Comment (sfackler) [inline phf_shared/src/lib.rs]: Hmm, this seems fine for now - hopefully that impl doesn't change.

**Comment by Ryman** [comment]
@sfackler Can you release a new version when you get a chance?

**Comment by sfackler** [comment]
Done

**Pull request merged**